### PR TITLE
feat(docs): add pwa with auto update strategy

### PR DESCRIPTION
**Pull request by userquin**
This PR includes:
- fix a11y and SEO problems: there are anchors without `rel="noopener noreferrer"`
- pwa with auto update strategy: to include prompt strategy we need to switch to `@vue/theme` or add a custom theme
- png icons added to use them on the pwa manifest
- netlify stuff to use correct cache and also adding some security entries on `netlify.toml` (also added: we need to review it)
- added a few dev dependencies: `localhost-https` added to test pwa on local
- added some scripts on root and on docs to allow build and serve the docs using https (`localhost-https`)
- added `scripts/build-pwa.ts` module to rebuild the pwa once `vitepress` finish: also added `esno` to run the module => I run it using `esno` instead `esmo` because it never finish, I don't know what's happing with it

Pending tasks, will not be fixed here, see #580

### CODE REVIEW

**Comment by userquin** [comment]
To run locally just `pnpm run docs:https` from root , `localhost-https` will ask to create and install `ssl/tls` certificate.

If you don't want to test the pwa, just run `pnpm run docs:build && pnpm run docs:serve`  from root.

To test the pwa, try it on chrome canary, lighthouse is still broken on chromium based browsers.

I mean, to test pwa with lighthouse, the pwa works on any browser.

File: docs/netlify.toml
Comment (antfu) [inline docs/netlify.toml:17]: We have a root-level one. Is this neccessary?

File: docs/scripts/assets.ts
Comment (antfu) [inline docs/scripts/assets.ts:35]: Can we automate this? I am not so confident of duplicating (It would make future changes much harder)

File: docs/netlify.toml
Comment (userquin) [reply docs/netlify.toml:17]: Upps, I don't see it, sorry, we can use the top one adding the webmanifest entry.

Should we use 301 instead 302 on next redirection?
/cc @patak-dev

File: docs/scripts/assets.ts
Comment (userquin) [reply docs/scripts/assets.ts:35]: We can include some module with the sita data, including all stuff there: avatars, links, contributors, fonts...

I'll be back next Tuesday, we can merge and then change it, but proceed as you wish

File: docs/netlify.toml
Comment (antfu) [reply docs/netlify.toml:17]: I prefer 302

**Comment by userquin** [comment]
upps, I need to check offline support, it seems the avatars will not load.

**Comment by patak-dev** [comment]
Looks awesome @userquin!

The only detail I see is that avatar images are downscaled too much now and they are seen blurry on my monitor:
image

Compared to the current one:
image

Maybe we can use a bit higher resolution for them? Same for the team avatars, you can also see the diff with our sponsor images

File: docs/src/contributors.ts
Comment (Demivan) [inline docs/src/contributors.ts:64]: Copy-pasted from `sheremet-va` description :)

Comment (userquin) [reply docs/src/contributors.ts:64]: upps, sorry, updating

**Comment by userquin** [comment]
@patak-dev testing on local to increase size

File: docs/contributors.json
Comment (antfu) [inline docs/contributors.json:4]: Can't we just use github.com/<name>.png?

File: scripts/update-contributors.ts
Comment (antfu) [inline scripts/update-contributors.ts:19]: The size doesn't matter much, let's keep the codebase simple

File: docs/contributors.json
Comment (userquin) [reply docs/contributors.json:4]: yes, but there will be N + M client redirections (where N is the number of core team members and M is the number of contributors)

File: scripts/update-contributors.ts
Comment (userquin) [reply scripts/update-contributors.ts:19]: I don't think so, just compare these 2:

Your avatar using `https://github.com/antfu.png`

imagen

And your avatar using `https://github.com/antfu.png?size=125` or `https://avatars.githubusercontent.com/u/11247099?v=4&s=125`:

imagen

Comment (userquin) [reply scripts/update-contributors.ts:19]: 744KB

**Comment by userquin** [comment]
I don't know why it is failing, on my windows machine running `nr ci` the error is:

<details>
<summary><strong>error on test/runner.test.ts<strong></summary>

```shell
test/fails test:  × test/runner.test.ts > should fails > nested-suite.test.ts
test/fails test:    → Snapshot `should fails > nested-suite.test.ts > nested-suite.test.ts 1` mismatched
test/fails test:  FAIL  test/runner.test.ts > should fails > nested-suite.test.ts
test/fails test: AssertionError: Snapshot `should fails > nested-suite.test.ts > nested-suite.test.ts 1` mismatched
test/fails test:  ❯ SnapshotClient.assert ../../packages/vitest/dist/entry.js:976:53
test/fails test:  ❯ Proxy.<anonymous> ../../packages/vitest/dist/entry.js:1042:27
test/fails test:  ❯ Proxy.methodWrapper ../../node_modules/.pnpm/chai@4.3.6/node_modules/chai/lib/chai/utils/addMethod.js:57:25
test/fails test:  ❯ test/runner.test.ts:31:18
test/fails test:      29|         .find(i => i.includes('Error: '))
test/fails test:      30|         ?.trim()
test/fails test:      31|       expect(msg).toMatchSnapshot(file)
test/fails test:        |                  ^
test/fails test:      32|     }, 10000)
test/fails test:      33|   }
test/fails test:  ❯ async runTest ../../packages/vitest/dist/entry.js:1695:5
test/fails test:  ❯ async runSuite ../../packages/vitest/dist/entry.js:1747:13
test/fails test:  ❯ async runSuite ../../packages/vitest/dist/entry.js:1747:13
test/fails test:  ❯ async runSuites ../../packages/vitest/dist/entry.js:1775:5
test/fails test:  ❯ async startTests ../../packages/vitest/dist/entry.js:1787:3
test/fails test:   - Expected   "AssertionError: expected true to equal false"
test/fails test:   + Received   "AssertionError: expected true to be false // Object.is equality"
test/fails test: ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯
test/fails test:  Snapshots  1 failed
test/fails test: Test Files  1 failed (1)
test/fails test:      Tests  1 failed | 3 passed (4)
test/fails test:       Time  13.68s (in thread 11.52s, 118.78%)
test/fails test: Failed
undefined
F:\work\projects\quini\GitHub\antfu\vitest-clean\test\fails:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @vitest/test-fails@ test: `vitest`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
```
</details>

**Comment by Demivan** [comment]
> I don't know why it is failing, on my windows machine running nr ci the error is:

Looks like snapshot needs updating

**Comment by userquin** [comment]
I've disabled the PWA, it seems there is some problem with the sw and the images on home page: we only need to change `pwaDisabled` on `docs/docs-data.ts` to enable it again.

I've changed all avatars (core team members and contributors) and sponsors images to be `loading="lazy"` if the PWA is disabled, I need to check what's happenning.

With these latest changes, the docs will work as the current docs deployed on netlify with some performance added: lazy load images.

I think we can merge with the PWA disabled and then fix the images problem: we'll be able also to start adding the new vitepress theme.

<details>
<summary>To test the docs on the preview you will need to remove the service worker previously installed:</summary>

- Open `dev tools` (`Option + ⌘ + J` on `macOS`, `Shift + CTRL + J` on `Windows/Linux`)
- Go to `Application > Storage`, you should check following checkboxes:
  - Application: [x] Unregister service worker
  - Storage: [x] Local and session storage
  - Cache: [x] Cache storage and [x] Application cache
- Click on `Clear site data` button
- Go to `Application > Service Workers` and check the current `service worker` is missing or has the state `deleted`.
</details>

**Comment by userquin** [comment]
reverting last commit, pnpm problem when updating, points to wrong scripts when using nvm

**Comment by userquin** [comment]
Since the sponsors images can be changed I revert fixed size

**Comment by userquin** [comment]
https://github.com/GoogleChrome/workbox/issues/3023

**Comment by userquin** [comment]
The PWA is disabled until we fix the contributors avarts CORS problem: to enable PWA we only need to change `pwaDisabled = false` on `docs/docs-data.ts` module.

**Review by patak-dev — APPROVED** [review]

**Comment by patak-dev** [comment]
Images are working well, I was seeing them blurry because the old SW was getting in the middle. I think we should merge this one

File: docs/src/contributors.ts
Comment (Aslemammad) [inline docs/src/contributors.ts:51]: That's amazing, Could you make my name here `Mohammad Bagher` ?

Comment (userquin) [reply docs/src/contributors.ts:51]: Change it yourself here editing the file and commiting changes to the brabch

Comment (userquin) [reply docs/src/contributors.ts:51]: If you dont want to change yourself just wait tomorrow or maybe @patak-dev can change it before merging into main

**Pull request merged**
